queries/js/connectHub.js

```diff
@@ -0,0 +1,90 @@
+//const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
```

Comment:
I suggest that you add "use strict" to the top of all the JS documents and run a linter on them (e.g. jshint) to make sure everything is top notch.